lib/index.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+// external modules
+const Assembler = require('stream-json/utils/Assembler');
+const assert = require('assert-plus');
+const JSONParseStream = require('stream-json/Combo');
+const through2 = require('through2');
+const stringifyStreamFactory = require('json-stream-stringify');
+
+
+/**
+ * create a JSON.parse that uses a stream interface. the underlying stream-json
+ * module used to do the parsing is actually a combo stream that has 3 internal
+ * streams:
+ *   * Parser (reading in raw strings)
+ *   * Streamer (converts token into SAX-like event stream)
+ *   * Packer (which assembles individual JS primitives from parsed chunks).
+ *
+ * Finally, an "Assembler" class is used to reconstruct the full POJO by
+ * assembling the individual JSON chunks. The caveat is that the process must
+ * be able to hold the fully reconstructed object in memory.
+ *
+ * However, since the assembler is not stream and works outside the streams
+ * themselves, we must create a wrapper stream that that encapsulates the
+ * execution of the 3 internal streams an the assembler. that way external
+ * consumers can expect to work with the standard stream events like 'data or
+ * 'end'. the wrapper stream can be thought of as an accumulator of the JSON
+ * chunks that will emit the fully constructed POJO once it is complete.
```

Comment:
So is this a real stream or not? might want to explicitly state which it is. If it's not a real stream but merely "looks" like a stream, it won't respect back pressure, which is important to communicate to your consumers :)

Reply:
The assembler is not, but that's the reason this module exists, to abstract over that and return a streams interface. Will make that clearer